docker/docker_image_dest.go

```diff
@@ -341,39 +341,52 @@ func (d *dockerImageDestination) TryReusingBlobWithOptions(ctx context.Context,
+				requiredCompression := "nil"
+				if compressionAlgorithm != nil {
+					requiredCompression = compressionAlgorithm.Name()
+				}
+				logrus.Debugf("Ignoring candidate blob %s as reuse candidate due to compression mismatch ( %s vs %s ) in %s", candidate.Digest.String(), options.RequiredCompression.Name(), requiredCompression, candidateRepo.Name())
+				continue
+			}
+			if candidate.CompressorName != blobinfocache.Uncompressed {
+				logrus.Debugf("Trying to reuse cached location %s compressed with %s in destination repo %s", candidate.Digest.String(), candidate.CompressorName, candidateRepo.Name())
+			} else {
+				logrus.Debugf("Trying to reuse blob with cached digest %s in destination repo %s", candidate.Digest.String(), candidateRepo.Name())
+			}
+			// Sanity checks:
+			if reference.Domain(candidateRepo) != reference.Domain(d.ref.ref) {
+				// OCI distribution spec 1.1 allows mounting blobs without specifying the source repo
+				// (the "from" parameter); in that case we might try to use these candidates as well.
+				//
+				// OTOH that would mean we can’t do the “blobExists” check, and if there is no match
+				// we could get an upload request that we would have to cancel.
+				logrus.Debugf("... Internal error: domain %s does not match destination %s", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
+				continue
 			}
-			logrus.Debugf("Ignoring candidate blob %s as reuse candidate due to compression mismatch ( %s vs %s ) in %s", candidate.Digest.String(), options.RequiredCompression.Name(), requiredCompression, candidateRepo.Name())
-			continue
-		}
-		if candidate.CompressorName != blobinfocache.Uncompressed {
-			logrus.Debugf("Trying to reuse cached location %s compressed with %s in %s", candidate.Digest.String(), candidate.CompressorName, candidateRepo.Name())
 		} else {
-			logrus.Debugf("Trying to reuse cached location %s with no compression in %s", candidate.Digest.String(), candidateRepo.Name())
-		}
-
-		// Sanity checks:
-		if reference.Domain(candidateRepo) != reference.Domain(d.ref.ref) {
-			// OCI distribution spec 1.1 allows mounting blobs without specifying the source repo
-			// (the "from" parameter); in that case we might try to use these candidates as well.
-			//
-			// OTOH that would mean we can’t do the “blobExists” check, and if there is no match
-			// we could get an upload request that we would have to cancel.
-			logrus.Debugf("... Internal error: domain %s does not match destination %s", reference.Domain(candidateRepo), reference.Domain(d.ref.ref))
-			continue
+			if candidate.CompressorName != blobinfocache.Uncompressed {
+				logrus.Debugf("Trying to reuse cached location %s compressed with %s in destination repo %s", candidate.Digest.String(), candidate.CompressorName, "unknown")
+			} else {
+				logrus.Debugf("Trying to reuse blob with cached digest %s in destination repo %s", candidate.Digest.String(), "unknown")
```

Comment:
Similarly, these two log messages should match apart from compression (and directly correspond to the known-location messages).

(Also, `"destination repo %s", "unknown"` seems inelegant. Maybe this could say something like “… to reuse a blob %{digest} with no location match, checking current repo”?)